pkg/encoding/v1/encoding_test.go

```diff
@@ -21,38 +21,40 @@ func UriPathAddrFromString(s string) *PathRegex {
 
 func TestPortMapping_UnmarshalYAML(t *testing.T) {
 	tests := []struct {
+		Name    string
 		Succeed bool
 		RawPort string
 		Port    *PortMapping
 	}{
-		{true, "5000", &PortMapping{ContainerPort: 5000, ServicePort: 5000}},
-		{true, "5000:80", &PortMapping{ContainerPort: 5000, ServicePort: 80}},
-		{true, "", &PortMapping{}}, // UnmarshalYAML won't be even called for empty strings
-		{false, "x5000", nil},
-		{false, "5000:", nil},
-		{false, "x5000:", nil},
-		{false, ":80", nil},
-		{false, ":80x", nil},
-		{false, "x:80x", nil},
-		{false, "x:80", nil},
-		{false, ":8080:80", nil},
-		{false, "x:8080:80", nil},
-		{false, "x:x8080:x80", nil},
-		{false, ":8080:x80", nil},
-		{false, ":8080:80:", nil},
-		{false, "8080:80:", nil},
-		{false, "8080:80::", nil},
-		{false, ":", nil},
-		{false, "::", nil},
-		{false, ":::", nil},
-		{false, "::::", nil},
-		{false, ":::::", nil},
-		{false, "::1:80", nil},
-		{false, "::1:8080:80", nil},
-		{false, "::1:8080:80:", nil},
+		{"Only ContainerPort", true, "5000", &PortMapping{ContainerPort: 5000, ServicePort: 5000}},
+		{"ContainerPort:ServicePort", true, "5000:80", &PortMapping{ContainerPort: 5000, ServicePort: 80}},
+		{"Empty Portmapping", true, "", &PortMapping{}}, // UnmarshalYAML won't be even called for empty strings
+		{"Failed Portmapping case-1", false, "x5000", nil},
+		{"Failed Portmapping case-2", false, "5000:", nil},
+		{"Failed Portmapping case-3", false, "x5000:", nil},
+		{"Failed Portmapping case-4", false, ":80", nil},
+		{"Failed Portmapping case-5", false, ":80x", nil},
+		{"Failed Portmapping case-6", false, "x:80x", nil},
+		{"Failed Portmapping case-7", false, "x:80", nil},
+		{"Failed Portmapping case-8", false, ":8080:80", nil},
+		{"Failed Portmapping case-9", false, "x:8080:80", nil},
+		{"Failed Portmapping case-10", false, "x:x8080:x80", nil},
+		{"Failed Portmapping case-11", false, ":8080:x80", nil},
+		{"Failed Portmapping case-12", false, ":8080:80:", nil},
+		{"Failed Portmapping case-13", false, "8080:80:", nil},
+		{"Failed Portmapping case-14", false, "8080:80::", nil},
+		{"Failed Portmapping case-15", false, ":", nil},
+		{"Failed Portmapping case-16", false, "::", nil},
+		{"Failed Portmapping case-17", false, ":::", nil},
+		{"Failed Portmapping case-18", false, "::::", nil},
+		{"Failed Portmapping case-19", false, ":::::", nil},
+		{"Failed Portmapping case-20", false, "::1:80", nil},
+		{"Failed Portmapping case-21", false, "::1:8080:80", nil},
+		{"Failed Portmapping case-22", false, "::1:8080:80:", nil},
 	}
 
 	for _, tt := range tests {
+		t.Log("Test case: ", tt.Name)
```

Comment:
How about using `t.Run(tt.name, func(t *testing.T)`, you can find the references in other test cases in the same file.